tests/test_metadata.py

```diff
@@ -0,0 +1,78 @@
+import pytest
+
+
+class TestMetadata(object):
+
+    def test_invalid_name(self, w_checkout):
+        # TODO: It's a value error in dataset and branch
+        with pytest.raises(SyntaxError):
+            w_checkout.metadata.add('a a', 'b')
+
+    def test_name_conflict(self, w_checkout):
+        w_checkout.metadata.add('a', 'b')
+        w_checkout.commit()
+        # TODO: shouldn't it raise a warning atleast?
+        w_checkout.metadata.add('a', 'c')
+        w_checkout.commit('second time')
+        assert w_checkout.metadata.get('a') == 'c'
+
+    def test_other_datatype(self, w_checkout):
+        # TODO: should raise a proper error
+        with pytest.raises(AttributeError):
+            w_checkout.metadata.add(1, '1')
+        with pytest.raises(AttributeError):
+            w_checkout.metadata.add('1', 1)
+
+    def test_read_only_mode(self, written_repo):
+        co = written_repo.checkout()
+        # TODO: should be permissionerror
```

Comment:
Good catch. Please create an issue and reference it to this comment!

Probably will be fixed by a solution to https://github.com/tensorwerk/hangar-py/pull/25#discussion_r281058815